server/src/test/java/org/elasticsearch/common/util/concurrent/ListenableFutureTests.java

```diff
@@ -142,7 +142,7 @@ public void testAddedListenersReleasedOnCompletion() {
         }
         reachabilityChecker.ensureUnreachable();
 
-        future.addListener(reachabilityChecker.register(ActionListener.wrap(() -> {})));
+        future.addListener(reachabilityChecker.register(ActionListener.running(() -> {})));
```

Comment:
Another example here

Reply:
This one is right, we need a new instance because we're checking reachability, and `noop()` returns a static field which is therefore always reachable.